akka-samples/akka-docs-java-lambda/src/test/java/docs/stream/GraphCyclesDocTest.java

```diff
@@ -0,0 +1,162 @@
+package docs.stream;
+
+import java.util.Arrays;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import akka.actor.ActorSystem;
+import akka.stream.FlowMaterializer;
+import akka.stream.OverflowStrategy;
+import akka.stream.javadsl.Broadcast;
+import akka.stream.javadsl.Concat;
+import akka.stream.javadsl.Flow;
+import akka.stream.javadsl.FlowGraph;
+import akka.stream.javadsl.Merge;
+import akka.stream.javadsl.MergePreferred;
+import akka.stream.javadsl.Sink;
+import akka.stream.javadsl.Source;
+import akka.stream.javadsl.Zip2With;
+import akka.stream.javadsl.ZipWith;
+import akka.testkit.JavaTestKit;
+
+
+public class GraphCyclesDocTest {
+
+  static ActorSystem system;
+  
+
+  @BeforeClass
+  public static void setup() {
+    system = ActorSystem.create("GraphCyclesDocTest");
+  }
+
+  @AfterClass
+  public static void tearDown() {
+    JavaTestKit.shutdownActorSystem(system);
+    system = null;
+  }
+  
+  final FlowMaterializer mat = FlowMaterializer.create(system);
+  
+  final Source<Integer> source = Source.from(Arrays.asList(1, 2, 3, 4, 5));
+  
+  @Test
+  public void demonstrateDeadlockedCycle() {
+    //#deadlocked
+    // WARNING! The graph below deadlocks!
+    final Merge<Integer> merge = Merge.create();
+    final Broadcast<Integer> bcast = Broadcast.create();
+    
+    final Flow<Integer, Integer> printFlow = 
+      Flow.<Integer>empty().map(s -> {
```

Comment:
Wouldn't `Flow.of(Integer.class)` look a bit nicer for java people? I personally don't like the `Class.<T>method` syntax a lot.